grizzly_extras/async_message/mq/rfh2.py

```diff
@@ -0,0 +1,157 @@
+import gzip
+import time
+import xml.etree.ElementTree as ET
+from struct import pack, unpack
+from typing import List, Optional
+
+try:
+    import pymqi
+except:
+    from grizzly_extras import dummy_pymqi as pymqi
+
+# Basic MQ RFH2 support with gzip compression
+# See https://www.ibm.com/docs/en/ibm-mq/9.1?topic=mqi-mqrfh2-rules-formatting-header-2
+
+
+class Rfh2Decoder(object):
+    @classmethod
+    def is_rfh2(cls, message: bytes) -> bool:
+        try:
+            parts = unpack('<4c l', message[0:8])
+            struc_id = b''.join(list(parts[0:4])).decode('ascii')
+            version = parts[4]
+            return struc_id == 'RFH ' and version == 2
+        except:
+            return False
+
+    def __init__(self, message: bytes) -> None:
+        self._parse_header(message)
+        self._parse_name_values()
+
+    def _parse_header(self, message: bytes) -> None:
+        header_length = 36
+        try:
+            parts = unpack('<4c l l l l 8c l l', message[0:header_length])
+            self.struc_id = b''.join(list(parts[0:4])).decode('ascii')
+            self.version = parts[4]
+            self.struc_length = parts[5]
+            self.encoding = parts[6]
+            self.charset = parts[7]
+            self.fmt = b''.join(list(parts[8:16])).decode('ascii')
+            self.flags = parts[16]
+            self.name_value_ccsid = parts[17]
+            self.name_values = message[header_length:self.struc_length]
```

Comment:
does everything between line 35-43 really need to be class variables?

Reply:
Nope! :)